### DISCUSSION TITLE
repeating Autohold

### DISCUSSION THREAD

**Discussion by FrontalLobotomyGit** [arkypita/LaserGRBL#2144] [Q&A]
Started a project today and lasergrbl will autohold every 10 seconds or so.  

There were a few odd things in the gcode log, but I didn't see them coincide with the hold.   After about 10 tries to continue, I gave up.   

Several MSG:Restoring spindle (blue)
and intermittent "Bad number format" (red)
i.e. G0 X12.1 Y63.7 S0
Y112.1 S468 (part of a G1 group of commands maybe?), but no error on the prior command Y112 S470 and the following Y112.1 S468

There were only 2 or 3 red error messages, not nearly enough to match the number of times I had to continue.  

Any ideas why this is happening?

**Comment by FrontalLobotomyGit** [comment]
Session Log, if it helps:
22/10/2023 16:17:42.949	Program     	------- LaserGRBL v5.4.0 [64bit] START -------
22/10/2023 16:17:46.968	Update DB   	Material DB is up to date
22/10/2023 16:19:08.646	OpenFile    	Open C:\Users\\OneDrive\projects\2023 Solstice\RubenClock975.png
22/10/2023 16:28:25.607	OpenFile    	Open C:\Users\\OneDrive\projects\2023 Solstice\RubenClock975.png
22/10/2023 16:32:41.912	OpenFile    	Open C:\Users\\OneDrive\projects\2023 Solstice\RubenClock975.png
22/10/2023 16:35:53.688	SetStatus   	Machine status [Connecting]
22/10/2023 16:35:53.696	OpenCom     	Open COM5 @ 921600 baud  (UsbSerial)
22/10/2023 16:35:53.703	SendImmediate	Send Immediate Command [0x18]
22/10/2023 16:35:53.755	VersionInfo 	Detected Grbl v1.1f
22/10/2023 16:35:53.968	SetStatus   	Machine status [Idle]
22/10/2023 16:35:54.520	EnlargeBuffer	Buffer size changed to 128 [True]
22/10/2023 16:36:13.770	SetStatus   	Machine status [Home]
22/10/2023 16:36:13.965	SetStatus   	Machine status [Idle]
22/10/2023 16:36:17.215	SetStatus   	Machine status [Run]
22/10/2023 16:36:17.964	SetStatus   	Machine status [Idle]
22/10/2023 16:37:30.464	SetStatus   	Machine status [Run]
22/10/2023 16:38:25.214	SetStatus   	Machine status [Idle]
22/10/2023 16:38:37.785	EnqueueProgram	Push Header
22/10/2023 16:38:37.785	EnqueueProgram	Push File, 84413 lines
22/10/2023 16:38:37.802	EnqueueProgram	Running program, 84413 lines
22/10/2023 16:38:37.964	SetStatus   	Machine status [Run]
22/10/2023 16:38:40.464	SetStatus   	Machine status [AutoHold]
22/10/2023 16:40:13.293	SendImmediate	Send Immediate Command [0x7E]
22/10/2023 16:40:13.465	SetStatus   	Machine status [Run]
22/10/2023 16:40:17.715	SetStatus   	Machine status [AutoHold]
22/10/2023 16:40:19.822	SendImmediate	Send Immediate Command [0x7E]
22/10/2023 16:40:19.965	SetStatus   	Machine status [Run]
22/10/2023 16:40:31.213	SetStatus   	Machine status [AutoHold]
22/10/2023 16:40:33.580	SendImmediate	Send Immediate Command [0x7E]
22/10/2023 16:40:33.714	SetStatus   	Machine status [Run]
22/10/2023 16:40:48.964	SetStatus   	Machine status [AutoHold]
22/10/2023 16:40:51.109	SendImmediate	Send Immediate Command [0x7E]
22/10/2023 16:40:51.213	SetStatus   	Machine status [Run]
22/10/2023 16:40:56.713	SetStatus   	Machine status [AutoHold]
22/10/2023 16:40:58.792	SendImmediate	Send Immediate Command [0x7E]
22/10/2023 16:40:58.965	SetStatus   	Machine status [Run]
22/10/2023 16:41:04.982	SetStatus   	Machine status [AutoHold]
22/10/2023 16:41:06.686	SendImmediate	Send Immediate Command [0x7E]
22/10/2023 16:41:06.714	SetStatus   	Machine status [Run]
22/10/2023 16:41:14.213	SetStatus   	Machine status [AutoHold]
22/10/2023 16:41:16.335	SendImmediate	Send Immediate Command [0x7E]
22/10/2023 16:41:16.464	SetStatus   	Machine status [Run]
22/10/2023 16:41:22.713	SetStatus   	Machine status [AutoHold]
22/10/2023 16:41:24.510	SendImmediate	Send Immediate Command [0x7E]
22/10/2023 16:41:24.714	SetStatus   	Machine status [Run]
22/10/2023 16:41:30.214	SetStatus   	Machine status [AutoHold]
22/10/2023 16:41:32.245	SendImmediate	Send Immediate Command [0x7E]
22/10/2023 16:41:32.465	SetStatus   	Machine status [Run]
22/10/2023 16:41:34.469	SetStatus   	Machine status [AutoHold]
22/10/2023 16:41:35.694	SendImmediate	Send Immediate Command [0x7E]
22/10/2023 16:41:35.965	SetStatus   	Machine status [Run]
22/10/2023 16:41:37.463	SetStatus   	Machine status [AutoHold]
22/10/2023 16:41:46.982	SendImmediate	Send Immediate Command [0x7E]
22/10/2023 16:41:47.213	SetStatus   	Machine status [Run]
22/10/2023 16:41:50.716	SetStatus   	Machine status [AutoHold]
22/10/2023 16:41:52.104	SendImmediate	Send Immediate Command [0x7E]
22/10/2023 16:41:52.214	SetStatus   	Machine status [Run]
22/10/2023 16:41:53.216	SetStatus   	Machine status [AutoHold]
22/10/2023 16:41:54.282	SendImmediate	Send Immediate Command [0x7E]
22/10/2023 16:41:54.464	SetStatus   	Machine status [Run]
22/10/2023 16:41:57.214	SetStatus   	Machine status [AutoHold]
22/10/2023 16:42:14.073	SendImmediate	Send Immediate Command [0x7E]
22/10/2023 16:42:14.214	SetStatus   	Machine status [Run]
22/10/2023 16:42:17.464	SetStatus   	Machine status [AutoHold]
22/10/2023 16:42:19.028	SendImmediate	Send Immediate Command [0x7E]
22/10/2023 16:42:19.214	SetStatus   	Machine status [Run]
22/10/2023 16:42:26.214	SetStatus   	Machine status [AutoHold]
22/10/2023 16:42:27.535	SendImmediate	Send Immediate Command [0x7E]
22/10/2023 16:42:27.714	SetStatus   	Machine status [Run]
22/10/2023 16:42:35.964	SetStatus   	Machine status [AutoHold]
22/10/2023 16:42:37.283	SendImmediate	Send Immediate Command [0x7E]
22/10/2023 16:42:37.464	SetStatus   	Machine status [Run]
22/10/2023 16:42:40.464	SetStatus   	Machine status [AutoHold]
22/10/2023 16:42:42.355	SendImmediate	Send Immediate Command [0x7E]
22/10/2023 16:42:42.464	SetStatus   	Machine status [Run]
22/10/2023 16:42:48.463	SetStatus   	Machine status [AutoHold]
22/10/2023 16:43:08.790	SendImmediate	Send Immediate Command [0x7E]
22/10/2023 16:43:08.964	SetStatus   	Machine status [Run]
22/10/2023 16:43:28.964	SetStatus   	Machine status [AutoHold]
22/10/2023 16:43:31.184	SendImmediate	Send Immediate Command [0x7E]
22/10/2023 16:43:31.216	SetStatus   	Machine status [Run]
22/10/2023 16:43:34.714	SetStatus   	Machine status [AutoHold]
22/10/2023 16:43:36.264	SendImmediate	Send Immediate Command [0x7E]
22/10/2023 16:43:36.464	SetStatus   	Machine status [Run]
22/10/2023 16:43:44.463	SetStatus   	Machine status [AutoHold]
22/10/2023 16:43:46.041	SendImmediate	Send Immediate Command [0x7E]
22/10/2023 16:43:46.214	SetStatus   	Machine status [Run]
22/10/2023 16:43:47.214	SetStatus   	Machine status [AutoHold]
22/10/2023 16:43:48.688	SendImmediate	Send Immediate Command [0x7E]
22/10/2023 16:43:48.715	SetStatus   	Machine status [Run]
22/10/2023 16:43:52.215	SetStatus   	Machine status [AutoHold]
22/10/2023 16:43:53.882	SendImmediate	Send Immediate Command [0x7E]
22/10/2023 16:43:53.965	SetStatus   	Machine status [Run]
22/10/2023 16:44:04.464	SetStatus   	Machine status [AutoHold]
22/10/2023 16:44:26.526	ManualAbort 	Program aborted by user action!
22/10/2023 16:44:26.527	Issue detector	ManualAbort [9,-1,-1]

**Reply by juvenalis549** [reply]
I'm facing the same problem. Did you find a solution

**Reply by arkypita** [reply]
AutoHold is the state of the machine going in hold (paused) state by its own decision. This behavior is not related to LaserGRBL but by the machine, so check from the manual or from the seller what cases the machine can go in hold (ie  overheating protection, safety features or something similar)

**Comment by juvenalis549** [comment]
What about a message on gbrl showing: Restoring spindle?
